stdlib/public/Reflection/TypeRefBuilder.cpp

```diff
@@ -361,150 +363,150 @@ TypeRefBuilder::getClosureContextInfo(RemoteRef<CaptureDescriptor> CD) {
 /// Dumping reflection metadata
 ///
 
-void
-TypeRefBuilder::dumpTypeRef(RemoteRef<char> MangledName,
-                            FILE *file, bool printTypeName) {
+void TypeRefBuilder::dumpTypeRef(RemoteRef<char> MangledName,
+                                 std::ostream &stream, bool printTypeName) {
   auto DemangleTree = demangleTypeRef(MangledName);
   auto TypeName = nodeToString(DemangleTree);
-  fprintf(file, "%s\n", TypeName.c_str());
+  stream << TypeName << "\n";
   auto Result = swift::Demangle::decodeMangledType(*this, DemangleTree);
   clearNodeFactory();
   if (Result.isError()) {
     auto *Error = Result.getError();
     char *ErrorStr = Error->copyErrorString();
     auto str = getTypeRefString(MangledName);
-    fprintf(file, "!!! Invalid typeref: %s - %s\n",
-            std::string(str.begin(), str.end()).c_str(), ErrorStr);
+    stream << "!!! Invalid typeref: " << str.str() << " - " << ErrorStr << "\n";
     Error->freeErrorString(ErrorStr);
     return;
   }
   auto TR = Result.getType();
-  TR->dump(file);
-  fprintf(file, "\n");
+  TR->dump(stream);
+  stream << "\n";
 }
 
-void TypeRefBuilder::dumpFieldSection(FILE *file) {
+void TypeRefBuilder::dumpFieldSection(std::ostream &stream) {
   for (const auto &sections : ReflectionInfos) {
     for (auto descriptor : sections.Field) {
       auto TypeDemangling =
-        demangleTypeRef(readTypeRef(descriptor, descriptor->MangledTypeName));
+          demangleTypeRef(readTypeRef(descriptor, descriptor->MangledTypeName));
       auto TypeName = nodeToString(TypeDemangling);
       clearNodeFactory();
-      fprintf(file, "%s\n", TypeName.c_str());
+      stream << TypeName.c_str() << "\n";
       for (size_t i = 0; i < TypeName.size(); ++i)
-        fprintf(file, "-");
-      fprintf(file, "\n");
+        stream << "-";
+      stream << "\n";
       for (auto &fieldRef : *descriptor.getLocalBuffer()) {
         auto field = descriptor.getField(fieldRef);
         auto fieldName = getTypeRefString(readTypeRef(field, field->FieldName));
-        fprintf(file, "%*s", (int)fieldName.size(), fieldName.data());
+        auto oldFlags = stream.flags();
+        stream << std::fixed << std::setprecision((int)fieldName.size())
+               << fieldName.data();
+        stream.flags(oldFlags);
```

Comment:
C++ really makes this difficult, huh. I think `stream.write()` might be better here, even though it doesn't quite mean the same thing (since it won't stop at a NUL character).